File: files/200-299/272/CH-272.py
import pandas as pd

path = "200-299/272/CH-272 Find Value.xlsx"
df = pd.read_excel(path, usecols="B:D", skiprows=1, nrows=7)
test = sorted(pd.read_excel(path, usecols="F", skiprows=1, nrows=12).iloc[:, 0].tolist())

unique = sorted(
    v for c in df for i, v in df[c].items()
    if (df[c] == v).sum() == 1 and (df.loc[i] == v).sum() == 1
)

print(unique == test)  # True

File: files/200-299/272/CH-272.R
library(tidyverse)
library(readxl)

path = "files/200-299/272/CH-272 Find Value.xlsx"
input = read_excel(path, range = "B2:D9") 
test  = read_excel(path, range = "F2:F13") %>% pull() %>% sort()

result = map2(
  row(input), col(input),
  ~ {
    val = input[[.y]][.x]
    is_unique_row = sum(input[.x, ] == val) == 1
    is_unique_col = sum(input[[.y]] == val) == 1
    if (is_unique_row && is_unique_col) val else NULL
  }
) %>%
  compact() %>%
  unlist() %>%
  sort()

all.equal(result, test)
# [1] TRUE
